src/Jellyfin.Extensions/EnumerableExtensions.cs

```diff
@@ -55,4 +57,84 @@ public static IEnumerable<T> SingleItemAsEnumerable<T>(this T item)
     {
         yield return item;
     }
+
+    /// <summary>
+    /// Gets all flags from an enum.
+    /// </summary>
+    /// <param name="value">The enum.</param>
+    /// <returns>The IEnumerable{Enum} containing all flags.</returns>
+    public static IEnumerable<Enum> GetFlags(this Enum value)
+    {
+        return GetFlags(value, Enum.GetValues(value.GetType()).Cast<Enum>().ToArray());
+    }
+
+    /// <summary>
+    /// Gets all individual flags from an enum.
+    /// </summary>
+    /// <param name="value">The enum.</param>
+    /// <returns>The IEnumerable{Enum} containeing all individual flags.</returns>
+    public static IEnumerable<Enum> GetIndividualFlags(this Enum value)
+    {
+        return GetFlags(value, GetFlagValues(value.GetType()).ToArray());
+    }
+
+    private static IEnumerable<Enum> GetFlags(Enum value, Enum[] values)
```

Comment:
Just use the `.HasFlag()` function. Additionally, the result from `Enum.GetValues` should be cached.

Sample: https://github.com/jellyfin/jellyfin/blob/master/src/Jellyfin.Extensions/Json/Converters/JsonFlagEnumConverter.cs